services/autoscaling/src/simcore_service_autoscaling/utils/utils_ec2.py

```diff
@@ -73,12 +73,19 @@ def closest_instance_policy(
     ec2_instance: EC2InstanceType,
     resources: Resources,
 ) -> float:
-    if ec2_instance.cpus < resources.cpus or ec2_instance.ram < resources.ram:
+    if (
+        ec2_instance.resources.cpus < resources.cpus
+        or ec2_instance.resources.ram < resources.ram
+    ):
         return 0
     # compute a score for all the instances that are above expectations
     # best is the exact ec2 instance
-    cpu_ratio = float(ec2_instance.cpus - resources.cpus) / float(ec2_instance.cpus)
-    ram_ratio = float(ec2_instance.ram - resources.ram) / float(ec2_instance.ram)
+    cpu_ratio = float(ec2_instance.resources.cpus - resources.cpus) / float(
```

Comment:
what if cpus or ram = 0 ?

Reply:
So you use this as a preidcate for sorting

ec2_instance is the hw available and resources is what is needed, right?

if not enough resources, then the ratio is negative and you return >100
if exact match, then ratio is 0 and you return 100
if enough resources, then the ratio is positive and <1 and you return <100

I guess that explains the reverse below (can perhaps use a change in sign and save to computations)

Reply:
ok, I agree with your point (cpus or ram = 0). will add an assert. but basically this cannot happen since these are the resources of EC2 possible hardware

Reply:
for the rest I will keep it as it is, as it seems to work fairly well